src/main/java/graphql/schema/idl/TypeDefinitionRegistry.java

```diff
@@ -334,6 +354,7 @@ private void removeFromMap(Map source, String key) {
             return Optional.of(handleReDefinition(olderEntry, newEntry));
         } else {
             target.put(name, newEntry);
+            schemaParseOrder.addDefinition(newEntry);
```

Comment:
It's really hard, at least for me, to judge if we're calling `addDefinition` in all the correct places. Seems to me that this is quite fragile too, like if we change the code and start parsing types differently, we could easily forget to call `schemaParseOrder.addDefinition` in the right place.
Maybe it's because I'm not too familiar with the code, but just wanted to pointed that out.

Reply:
Yeah its a good point - 80% of the different types go through the same `define` or `defineExt` but there are these exceptions